tls/s2n_ktls.c

```diff
@@ -121,66 +120,49 @@ static S2N_RESULT s2n_ktls_get_io_mode(s2n_ktls_mode ktls_mode, int *tls_tx_rx_m
     return S2N_RESULT_OK;
 }
 
-#if defined(S2N_KTLS_SUPPORTED)
-S2N_RESULT s2n_ktls_init_aes128_gcm_crypto_info(struct s2n_connection *conn, s2n_ktls_mode ktls_mode,
-        struct s2n_key_material *key_material, struct tls12_crypto_info_aes_gcm_128 *crypto_info)
+static S2N_RESULT s2n_ktls_crypto_info_init(struct s2n_connection *conn, s2n_ktls_mode ktls_mode,
+        struct s2n_ktls_crypto_info *crypto_info)
 {
     RESULT_ENSURE_REF(conn);
-    RESULT_ENSURE_REF(conn->client);
-    RESULT_ENSURE_REF(conn->server);
-    RESULT_ENSURE_REF(key_material);
-    RESULT_ENSURE_REF(crypto_info);
-    RESULT_ENSURE_REF(conn->secure);
-    RESULT_ENSURE_REF(conn->secure->cipher_suite);
-    RESULT_ENSURE_REF(conn->secure->cipher_suite->record_alg);
-    const struct s2n_cipher *cipher = conn->secure->cipher_suite->record_alg->cipher;
-    RESULT_ENSURE_REF(cipher);
-
-    RESULT_ENSURE(cipher == &s2n_aes128_gcm, S2N_ERR_KTLS_UNSUPPORTED_CONN);
-    RESULT_ENSURE(conn->actual_protocol_version == S2N_TLS12, S2N_ERR_KTLS_UNSUPPORTED_CONN);
```

Comment:
Correct. This method previously assumed TLS1.2 (it set crypto_info->info.version to TLS_1_2_VERSION), but it no longer makes that assumption. It defers to the cipher.

Theoretically even if we removed the check from s2n_ktls_validate() we'd probably be fine, because only TLS1.2 ciphers have set_ktls_info right now.